lib/middlewares/responseGenerator.js

```diff
@@ -105,6 +105,30 @@ function blackButtonResponse(parsedBody, req) {
     return payload;
 }
 
+function blackButtonError(parsedBody, req) {
+    var payload,
+        details,
+        code,
+        errorType;
+
+    if (req.cbError) {
+        if (req.cbError.code && req.cbError.code !== 200) {
+            code = req.cbError.code;
+            errorType = 0;
+            details = '';
+        } else {
+            code = req.cbError.details.code || 'UNK';
```

Comment:
It should, as the `cbError` corresponds to a EntityGenericError in the library:
```
    EntityGenericError: function(id, type, details, code) {
        this.name = 'ENTITY_GENERIC_ERROR';
        this.message = 'Error accesing entity data for device: ' + id + ' of type: ' + type;
        this.details = details || {};
        this.code = code || 200;
    },
```
That takes care of creating an empty object if there isn't any. It doesn't hurt to add a check, anyway.